src/server/cluster/cluster_family.cc

```diff
@@ -589,6 +593,28 @@ void ClusterFamily::DflyClusterFlushSlots(CmdArgList args, ConnectionContext* cn
   return rb->SendOk();
 }
 
+void ClusterFamily::DflyClusterStartSlotMigration(CmdArgList args, ConnectionContext* cntx) {
+  SinkReplyBuilder* rb = cntx->reply_builder();
+
+  args.remove_prefix(1);  // Removes "START-SLOT-MIGRATION" subcmd string
+
+  CmdArgParser parser(args);
+  auto [host_ip, port] = parser.Next<std::string_view, uint32_t>();
+  std::vector<ClusterConfig::SlotRange> slots;
+  do {
+    auto [slot_start, slot_end] = parser.Next<uint32_t, uint32_t>();
+    slots.emplace_back(slot_start, slot_end);
+  } while (parser.HasNext());
+  (void)slots;
+  if (auto err = parser.Error(); err)
+    return (*cntx)->SendError(err->MakeReply());
+
+  ClusterSlotMigration node(std::string(host_ip), port);
```

Comment:
You will need to save the ClusterSlotMigration that is in process . Currently you just connect but later you will open multiple fibers that will replicate the data from source to target.

There can be several migration process from different nodes , you will need to save the state of all of them

Reply:
I will do it in the next step